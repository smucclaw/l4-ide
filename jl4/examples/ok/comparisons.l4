
#EVAL 3 < 5
#EVAL 3 < 4
#EVAL 3 < 3
#EVAL 3 < 2

#EVAL 3 LESS THAN 2
#EVAL 3 BELOW 2

#EVAL 3 <= 5
#EVAL 3 <= 4
#EVAL 3 <= 3
#EVAL 3 <= 2

#EVAL 3 AT MOST 2

#EVAL 5 > 3
#EVAL 4 > 3
#EVAL 3 > 3
#EVAL 2 > 3

#EVAL 3 GREATER THAN 2
#EVAL 3 ABOVE 2

#EVAL 5 >= 3
#EVAL 4 >= 3
#EVAL 3 >= 3
#EVAL 2 >= 3

#EVAL 2 AT LEAST 3

#EVAL 2 = 3
#EVAL 3 = 3

#EVAL 3 EQUALS 3

#EVAL "bar" < "car"
#EVAL "bar" < "bar"
#EVAL "bar" < "bas"

#EVAL "bar" <= "car"
#EVAL "bar" <= "bar"
#EVAL "bar" <= "bas"

#EVAL "bar" = "bar"
#EVAL "bar" = "car"
#EVAL "bar" = "Bar"

#EVAL TRUE < FALSE
#EVAL TRUE <= FALSE
#EVAL FALSE <= FALSE
#EVAL FALSE <= TRUE

#EVAL TRUE = TRUE
#EVAL TRUE = FALSE
