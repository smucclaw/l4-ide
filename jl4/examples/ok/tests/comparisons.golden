Parsing successful
Typechecking successful
Evaluation successful
comparisons.l4:2:1-12:
  TRUE
comparisons.l4:3:1-12:
  TRUE
comparisons.l4:4:1-12:
  FALSE
comparisons.l4:5:1-12:
  FALSE
comparisons.l4:7:1-20:
  FALSE
comparisons.l4:8:1-16:
  FALSE
comparisons.l4:10:1-13:
  TRUE
comparisons.l4:11:1-13:
  TRUE
comparisons.l4:12:1-13:
  TRUE
comparisons.l4:13:1-13:
  FALSE
comparisons.l4:15:1-18:
  FALSE
comparisons.l4:17:1-12:
  TRUE
comparisons.l4:18:1-12:
  TRUE
comparisons.l4:19:1-12:
  FALSE
comparisons.l4:20:1-12:
  FALSE
comparisons.l4:22:1-23:
  TRUE
comparisons.l4:23:1-16:
  TRUE
comparisons.l4:25:1-13:
  TRUE
comparisons.l4:26:1-13:
  TRUE
comparisons.l4:27:1-13:
  TRUE
comparisons.l4:28:1-13:
  FALSE
comparisons.l4:30:1-19:
  FALSE
comparisons.l4:32:1-12:
  FALSE
comparisons.l4:33:1-12:
  TRUE
comparisons.l4:35:1-17:
  TRUE
comparisons.l4:37:1-20:
  TRUE
comparisons.l4:38:1-20:
  FALSE
comparisons.l4:39:1-20:
  TRUE
comparisons.l4:41:1-21:
  TRUE
comparisons.l4:42:1-21:
  TRUE
comparisons.l4:43:1-21:
  TRUE
comparisons.l4:45:1-20:
  TRUE
comparisons.l4:46:1-20:
  FALSE
comparisons.l4:47:1-20:
  FALSE
comparisons.l4:49:1-19:
  FALSE
comparisons.l4:50:1-20:
  FALSE
comparisons.l4:51:1-21:
  TRUE
comparisons.l4:52:1-20:
  TRUE
comparisons.l4:54:1-18:
  TRUE
comparisons.l4:55:1-19:
  FALSE
